### DISCUSSION TITLE
bringing in environments from Earth Engine imagery

### DISCUSSION THREAD

**Discussion by Kinchie** [JoelForamitti/agentpy#11] [Ideas]
Firstly congratulations on developing great ABM features for Python, similar to Netlogo (which I am using currently). 

I'm an ecologist working on plant-herbivore systems, and bringing periodically updated spatial data  (eg on tree cover, grass biomass, burn state) into an ABM from free remote-sensing platforms (like Google Earth Engine - in .asc or appropriate format) would be invaluable. 

Any suggestions on how to do this in an AGENTPY model system?

**Comment by jofmi** [comment]
Hi! That sounds like an exciting project!

If you want to use a discrete spatial grid environment in your model, the next update will include a way to add attributes to each grid cell, so that could be used to map spatial data to each cell. You can already take a look at this extension here: https://github.com/JoelForamitti/agentpy/tree/locations

If you want to use a continuous space environment, I would store the spatial data as a seperate object in the model, and let agents retrieve data from that object. Let's say the spatial data is stored in `model.mydata` and has a method `get_tree_cover` that takes a coordinate vector as an input, agents could access it with `self.model.mydata.get_tree_cover(self.position())`.

If you are looking for a GIS environment, that does unfortunately not exist yet. If you would be interested to work on an extension for this, I'd be happy to support it!

Let me know if that helps :)

**Comment by jofmi** [comment]
Hi @Kinchie, since AgentPy v0.1.0, the Grid class has a method Grid.add_field() that can be used for cell attributes. Find more about it here: https://agentpy.readthedocs.io/en/latest/reference_grid.html#agentpy.Grid.add_field

If you had something different in mind, could you explain your application here in more detail, maybe with a simple example?

**Reply by ernst117** [reply]
@JoelForamitti  I'm also a plant ecologist and would love to see models in agentpy able to read in GIS files. Perhaps connecting agentpy to existing GIS python modules (such as those found here: https://automating-gis-processes.github.io/2016/Lesson1-Intro-Python-GIS.html) would be a good start?

**Reply by jofmi** [reply]
Hi @ernst117, sorry for the late reply! This can definitely be done, I have it on my list of possible extensions :)
